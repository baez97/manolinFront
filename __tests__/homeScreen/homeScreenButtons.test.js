// import { SCLAlert, SCLAlertButton } from 'react-native-scl-alert';
import React from 'react';
import renderer from 'react-test-renderer';
import GlobalButton from '../../components/homeComponents/globalButton';
import MidBlueButton from '../../components/homeComponents/midBlueButton';
import MidYellowButton  from '../../components/homeComponents/midYellowButton';


jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

function simulatePress(renderedObject) {
    renderedObject
        .props
        .onStartShouldSetResponder
        .__reactBoundContext
        .props
        .onPress();
}

describe('GlobalButton', () => {
    it('Renders correctly', () => {
        const rendered = renderer.create(<GlobalButton />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('OnPress method calls the onPressFn', () => {
        const mockedFn = jest.fn();
        const rendered = renderer.create(
            <GlobalButton onPressFn={mockedFn}/>
        ).toJSON();

        simulatePress(rendered);

        expect(mockedFn).toBeCalled();
    });
});

describe('MidBlueButton', () => {
    it('Renders correctly', () => {
        const rendered = renderer.create(<MidBlueButton />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('OnPress method calls the onPressFn', () => {
        const mockedFn = jest.fn(() => {});
        const rendered = renderer.create(
            <MidBlueButton onPressFn={mockedFn}/>
        ).toJSON();

        simulatePress(rendered);

        expect(mockedFn).toBeCalled();
    });
});

describe('MidYellowButton', () => {
    it('Renders correctly', () => {
        const rendered = renderer.create(<MidYellowButton />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('OnPress method calls the onPressFn', () => {
        const mockedFn = jest.fn(() => {});
        const rendered = renderer.create(
            <MidYellowButton onPressFn={mockedFn}/>
        ).toJSON();

        simulatePress(rendered);

        expect(mockedFn).toBeCalled();
    });
});


