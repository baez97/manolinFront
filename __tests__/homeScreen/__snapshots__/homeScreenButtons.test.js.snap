// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalButton Renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4279286482,
        4280914167,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "borderRadius": 35.5,
        "elevation": 15,
        "flexDirection": "row",
        "height": 212.5,
        "justifyContent": "space-evenly",
        "marginTop": 35,
        "paddingLeft": 17.5,
        "paddingRight": 17.5,
        "paddingTop": 17.5,
        "width": 712.5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 38.95833442608516,
          "marginTop": 17.5,
          "textAlign": "left",
        }
      }
    />
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 177,
          "width": 177,
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;

exports[`MidBlueButton Renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4279286482,
        4278481116,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 15,
        "height": 141.5,
        "justifyContent": "center",
        "marginTop": 20,
        "width": 336.5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 35.416667660079604,
          "textAlign": "left",
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;

exports[`MidYellowButton Renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4294954035,
        4294089034,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 15,
        "height": 141.5,
        "justifyContent": "center",
        "marginTop": 20,
        "width": 336.5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "#3f2606",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 35.416667660079604,
          "textAlign": "left",
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;
